src/source/ObjcGenerator.scala

```diff
@@ -417,6 +420,44 @@ class ObjcGenerator(spec: Spec) extends Generator(spec) {
       }
       w.wl
 
+      w.wl("- (NSDictionary *)toDict")
+      w.braced {
+        w.wl("#define _djinni_hide_null_(_o_) ((_o_)?(_o_):([NSNull null]))")
```

Comment:
I suggest to just unroll it. The generated code only needs to be locked at by people who are developing Djinni, it is not targetted at users. Furthermore, only fields that are marked as `optional` can ever be `nil`, so the repetition is not as bad as it seems.